MetaMask/metamask-extension
# feat: add useFirstPartyContractName hook and use it in useDisplayName

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23092?quickstart=1)

## **Description**
Add `useFirstPartyContractName` hook to find names for our first-party contracts based on their address and chainID. 

Integrated with `useDisplayName` to ensure Petnames recognize these contracts correctly.

**Why It Matters:**
Prior to Petnames, MetaMask displayed first-party contracts like the MetaMask Bridge as cryptic hex addresses. This wasn't very user-friendly. However, with the introduction of Petnames in version 11.11, these addresses will start appearing even more alarming due to the new color scheme and icons. Our update aims to make these contracts easily recognizable and less intimidating.

Before Petnames:

<img src="https://github.com/MetaMask/metamask-extension/assets/507015/1ce9a85e-43b0-4fe3-a2ba-67bcd858a738" width="300px"/>

With Petnames, before this PR:

<img src="https://github.com/MetaMask/metamask-extension/assets/507015/bdedae38-80dc-405c-80b4-ce1ad599bf57" width="300px"/>



## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2136

## **Manual testing steps**

1. Set the current chain to mainnet
2. Start a send transaction from the extension wallet
3. Put `0x0439e60F02a8900a951603950d8D4527f400C3f1` as the recipient
4. Click "Next"
5. On the confirmation screen, the recipient should be labelled as "MetaMask Bridge"

If you want to test other addresses and names on other chains, you can refer to [first-party-contracts.ts in this PR for a full list](https://github.com/MetaMask/metamask-extension/pull/23092/files#diff-68785145ed32475edef8bbc5869898d31b70a56a7005c43657bec065b7450c2e).

## **Screenshots/Recordings**



### **Before**

See description

### **After**

![image](https://github.com/MetaMask/metamask-extension/assets/507015/412ba102-cda0-42df-8062-df748ddc2178)

![image](https://github.com/MetaMask/metamask-extension/assets/507015/a18ddaa8-1c50-404e-b7c8-819c62444673)


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
